### PR TITLE
Fix/trigger webhook on bulk delete

### PR DESCRIPTION
**Pull request by Marc-Roig**
## 👋🏻  What does it do?

- Webhook was not triggered when doing a bulk delete.

## ❓ Why is it needed?

It was triggered when deleting a single entity but not for a bulk delete. 

## 🧪  How to test it?

1. Create your webhook. ( [example](https://scribehow.com/shared/Create_webhook__gNj4Ww6AQIOZBImzXBJMLg) )
You can use this site to test the webhook https://webhook.site/
2. Delete a single entity from any content type.
3. Webhook (entry.delete) should have been triggered
4. Bulk delete entities from any content type
5. One webhook (entry.delete) should have been triggered **for each entity deleted**.

## 🔗 Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/12290

### CODE REVIEW

**Comment by Marc-Roig** [comment]
About the components not being removed from the database, @derrickmehaffy  mentions in the [issue](https://github.com/strapi/strapi/issues/14039#issuecomment-1209570377) that we could modify the foreign key constrain to enable cascade delete.
@petersg83 we could have a discussion tomorrow about this.

File: packages/core/strapi/lib/services/entity-service/index.js [hunk 1]
```diff
@@ -261,8 +261,9 @@ const createDefaultImplementation = ({ strapi, db, eventHub, entityValidator })
       return null;
     }
 
-    await db.query(uid).delete({ where: { id: entityToDelete.id } });
+    // Delete entity components first, and then the entity
     await deleteComponents(uid, entityToDelete);
+    await db.query(uid).delete({ where: { id: entityToDelete.id } });
```
Comment (petersg83) [inline packages/core/strapi/lib/services/entity-service/index.js:266]: Why did you change the order? It was not working before?

File: packages/core/strapi/lib/services/entity-service/index.js [hunk 2]
```diff
@@ -276,7 +277,22 @@ const createDefaultImplementation = ({ strapi, db, eventHub, entityValidator })
     // select / populate
     const query = transformParamsToQuery(uid, wrappedParams);
 
-    return db.query(uid).deleteMany(query);
+    const entitiesToDelete = await db.query(uid).findMany({
+      ...query,
+    });
```
Comment (petersg83) [inline packages/core/strapi/lib/services/entity-service/index.js:282]: What about:
```js
    const entitiesToDelete = await db.query(uid).findMany(query);
```
?

File: packages/core/strapi/lib/services/entity-service/index.js (hunk 1 above)
Comment (Marc-Roig) [reply packages/core/strapi/lib/services/entity-service/index.js:266]: `deleteComponents` was querying for the uid entity:
```js
// In deleteComponents:
      const value = await strapi.query(uid).load(entityToDelete, attributeName);
```

the entity was already deleted at the time `deleteComponents` was executed.

File: packages/core/strapi/tests/api/basic-dp-compo.test.e2e.js [hunk 3]
```diff
@@ -161,6 +161,13 @@ describe('Core API - Basic + compo + draftAndPublish', () => {
     data.productsWithCompoAndDP.shift();
   });
 
+  describe('database state', () => {
+    test('components have been removed from the database', async () => {
+      const dbComponents = await strapi.db.query('default.compo').findMany({});
```
Comment (petersg83) [inline packages/core/strapi/tests/api/basic-dp-compo.test.e2e.js:166]: Let's see what the fun test run says but it may not work as the test database may have data left from the other tests

**Comment by petersg83** [comment]
Very nice!
Let's talk about the cascade delete, that may be an option. What do you think to separate this in 2 PR? One for the webhook and one for the component deletion in DB?

File: packages/core/strapi/tests/api/basic-dp-compo.test.e2e.js (hunk 3 above)
Comment (Marc-Roig) [reply packages/core/strapi/tests/api/basic-dp-compo.test.e2e.js:166]: Isn't the database pruned in each test file? 
Maybe I can add a query in the findMany referring to the component added in this test file then.

**Comment by Marc-Roig** [comment]
> et's talk about the cascade delete, that may be an option. What do you think to separate this in 2 PR? One for the webhook and one for the component deletion in DB?

Sure! What do you think if we leave the deleteComponents reorder & test as a temporary fix, and we work on the cascade delete in another PR?

File: packages/core/strapi/tests/api/basic-dp-compo.test.e2e.js (hunk 3 above)
Comment (petersg83) [reply packages/core/strapi/tests/api/basic-dp-compo.test.e2e.js:166]: I'm not sure, it would be nice to check and know. Js wrote the code

Comment (Marc-Roig) [reply packages/core/strapi/tests/api/basic-dp-compo.test.e2e.js:166]: I will add the query just in case then.

**Comment by Marc-Roig** [comment]
@petersg83  
I see an e2e failing with my current change: `packages/core/strapi/tests/lifecycles/before-delete.test.e2e.js` Exactly at line 94.

From there, I saw a PR made by @stafyniaksacha some months ago: https://github.com/strapi/strapi/pull/12653

He changed the order of deletion to allow developers to query the entity being deleted including the components on the beforeDelete hook.

So I think I will leave things as they were, and maybe we can have this in mind when planning the cascade delete.

Note: props to @stafyniaksacha  for the e2e tests that made me realize this :)
